provisioning/device/src/interfaces.ts

```diff
@@ -56,29 +69,50 @@ export interface PollingTransportHandlers {
   getErrorResult(result: any): any;
 }
 
+/**
+ * Public API used to access the ProvisioningDeviceClient object
+ */
 export interface RegistrationClient {
-  register(callback: (err?: Error) => void): void;
+  /**
+   * Register the device with the provisioning service
+   */
+  register(registrationId: string, forceRegistration: boolean, callback: (err?: Error, result?: any) => void): void;
+  /**
+   * Cancel the registration process if it is in progress.
+   */
   cancel(callback: (err?: Error) => void): void;
 }
 
+/**
+ * @private
+ */
 export interface TpmRegistrationInfo {
   registrationId: string;
   endorsementKey: string;
   storageRootKey: string;
 }
 
+/**
+ * Device configuration returned when registration using TPM is complete
+ */
 export interface TpmRegistrationResult {
   deviceId: string;
-  iotHubUri: string;
+  assignedHub: string;
   symmetricKey: string;
 }
 
+/**
+ * @private
+ */
 export interface TpmProvisioningTransport {
   getAuthenticationChallenge(registrationInfo: TpmRegistrationInfo, callback: (err: Error, tpmChallenge?: TpmChallenge) => void): void;
   register(registrationInfo: TpmRegistrationInfo, sasToken: string, callback: (err: Error, result?: TpmRegistrationResult) => void): void;
```

Comment:
>register [](start = 2, length = 8)

note for later: this is missing a forceRegistration parameter.

Reply:
I'm wondering if this should be inside the TpmRegistrationInfo structure.  I'm also wondering if I should create a parallel X509RegistrationInfo structure.

---
In reply to: [153352352](https://github.com/Azure/azure-iot-sdk-node/pull/159#discussion_r153352352) [](ancestors = 153352352)

Reply:
Rule of thumb for me is that it makes sense to "box" invidual arguments together in a class if multiple functions take the same list of arguments, or if we need to chain function calls to fully fill a structure (although the latter has a code smell i can't explain). if you're seeing this on the X509 side go for it.